### PR TITLE
Add image unit tests

### PR DESCRIPTION
**Pull request by ryantwolf**
## Description

Adds unit tests for all the image curation techniques.

## Usage

N/A
## Checklist

- [x] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [x] New or Existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: tests/image/classifiers/test_aesthetic.py [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+    assert classifier.pred_type == float
+    assert classifier.batch_size == 64
+    assert classifier.embedding_size == 768
+    assert classifier.model_path == "/custom/path/model.pth"
+
+
+# Test _get_default_model method
+@pytest.mark.gpu
+def test_get_default_model():
+    """Test that _get_default_model returns the correct path and downloads if needed."""
```
Comment (sarahyurick) [inline tests/image/classifiers/test_aesthetic.py:78]: Does this check the `if not os.path.exists(model_path)` block?

File: tests/image/classifiers/test_nsfw.py [hunk 2]
```diff
@@ -0,0 +1,551 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
```
Comment (sarahyurick) [inline tests/image/classifiers/test_nsfw.py:1]: Is `class Normalization` tested anywhere?

File: tests/image/classifiers/test_nsfw.py [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+    assert classifier.pred_type == float
+    assert classifier.batch_size == 64
+    assert classifier.embedding_size == 768
+    assert classifier.model_path == "/custom/path/model.pth"
+
+
+# Test _get_default_model method
+@pytest.mark.gpu
+def test_get_default_model():
+    """Test that _get_default_model returns the correct path and downloads if needed."""
```
Comment (sarahyurick) [inline tests/image/classifiers/test_nsfw.py:79]: Does this check the `if not os.path.exists(model_path)` block?

File: tests/image/classifiers/test_aesthetic.py [hunk 4]
```diff
@@ -0,0 +1,512 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
```
Comment (sarahyurick) [inline tests/image/classifiers/test_aesthetic.py:1]: Are there any tests for `class ImageClassifier`?

File: tests/image/test_image_text_pair_dataset.py [hunk 5]
```diff
@@ -0,0 +1,957 @@
...
+            dataset = ImageTextPairDataset.from_webdataset(temp_dataset_dir, "id")
+
+            # Verify the result
+            assert dataset.path == temp_dataset_dir
+            assert dataset.id_col == "id"
+            assert len(dataset.tar_files) == 1
+            assert dataset.tar_files[0].endswith("00000.tar")
+
+    @pytest.mark.gpu
+    def test_save_metadata(self, temp_dataset_dir):
```
Comment (sarahyurick) [inline tests/image/test_image_text_pair_dataset.py:308]: Add test for `if path is None`?

File: tests/image/test_image_text_pair_dataset.py [hunk 6]
```diff
@@ -0,0 +1,957 @@
...
+                        tar.extractfile(member).read()
+                        if tar.extractfile(member)
+                        else b""
+                    )
+                    content.append((member, member_content))
+
+            # Mock the eligible samples generator to return our real data
+            mock_get_eligible_samples.return_value = [
+                (metadata_df.to_pandas(), content)
+            ]
```
Comment (sarahyurick) [inline tests/image/test_image_text_pair_dataset.py:864]: Does this test cover the entire `_get_eligible_samples` function?

File: tests/image/embedders/test_timm.py [hunk 7]
```diff
@@ -0,0 +1,515 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
```
Comment (sarahyurick) [inline tests/image/embedders/test_timm.py:1]: Are there any tests for `class ImageEmbedder`?

File: tests/image/classifiers/test_aesthetic.py (hunk 1 above)
Comment (ryantwolf) [reply tests/image/classifiers/test_aesthetic.py:78]: Yes

File: tests/image/classifiers/test_nsfw.py (hunk 2 above)
Comment (ryantwolf) [reply tests/image/classifiers/test_nsfw.py:1]: Yes. It's tested in the `test_classifier_call_with_mock_dataset` at least. The weight values are gibberish, but its functionality is tested. I'm adding another test with fixed weights.

File: tests/image/classifiers/test_nsfw.py (hunk 3 above)
Comment (ryantwolf) [reply tests/image/classifiers/test_nsfw.py:79]: Yes

File: tests/image/classifiers/test_aesthetic.py (hunk 4 above)
Comment (ryantwolf) [reply tests/image/classifiers/test_aesthetic.py:1]: Since it's fully tested by the other classifiers I left it out.

File: tests/image/test_image_text_pair_dataset.py (hunk 6 above)
Comment (ryantwolf) [reply tests/image/test_image_text_pair_dataset.py:864]: There are two small branches left to cover, adding tests for them.

File: tests/image/embedders/test_timm.py (hunk 7 above)
Comment (ryantwolf) [reply tests/image/embedders/test_timm.py:1]: Same as the classifiers, I only test the subclasses.

**Review by sarahyurick — APPROVED** [review]
LGTM, thanks!
